### DISCUSSION TITLE
Xmake and tauri

### DISCUSSION THREAD

**Discussion by A2va** [xmake-io/xmake#3580] [Q&A]
Tauri and xmake

If you don't know what [Tauri](https://tauri.app/) is, it's similar to Electron, but it has a Rust backend instead of a Javascript one for Electron. Since, xmake supports Rust I try to build a Tauri app with it, with this I could integrate with a C/C++ library more easily.

So far, my `xmake.lua` is like this:
```lua
add_requires("cargo::xmake-tauri", {configs = {cargo_toml = path.join(os.projectdir(),"src-tauri",  "Cargo.toml")}})
target("tauri")
    add_files("src-tauri/src/main.rs")

    add_packages("cargo::xmake-tauri")
    if is_plat("windows", "mingw") then
        add_syslinks("windows", { public = true })
    end
    on_load(function (target) 
        local prj = os.projectdir()

        os.setenv("CARGO_MANIFEST_DIR", path.join(prj, "src-tauri"))
        os.setenv("OUT_DIR", path.join(prj, "build", "windows", "x64", "release")) -- it's temporary 

        os.setenv("CARGO_PKG_DESCRIPTION", "Example")
        os.setenv("CARGO_PKG_AUTHORS", "A2va")
    end)
```

But the link failed with the following error:
`note: LINK : fatal error LNK1181: cannot open input file 'windows.lib'`

From the Tauri website I have the necessary components in Visual Studio, I also tested with `npm run tauri dev` that it starts the app normally.

**Comment by waruqi** [comment]
windows.lib? where is it?

**Reply by A2va** [reply]
I don't know, windows.lib is suppose to be installed along the Windows SDK (https://stackoverflow.com/questions/59648046/unable-to-compile-rust-program-lnk1181-cannot-open-input-file-c-program-obj). But I don't know why it doesn't find it, and it's not the fault of `add_syslinks` (I try with and without the same error appears).

**Reply by A2va** [reply]
I have changed my approach and start [xmake-rs](https://github.com/A2va/xmake-rs) which is the same idea as cmake-rs. This will allow xmake to be invoked from a build script.
